src/pages/Settings/Settings.js

```diff
@@ -1,146 +1,182 @@
+              key={port[0] + port[1]}
+              label={createLabelfromSnakeCase(port[0])}
+              content={port[1]} />
+          })
+          : DEFAULT_PORT_NAMES.map(port => <SettingsRow
+            key={port}
+            label={port}
+            content={NOT_AVAILABLE} />)
+        }
+      </SettingsTable>
+    </section>
+
+    <UpdateSoftwareModal
+      settings={settings}
+      handleClose={() => setSoftwareUpdateVersion(null)}
+      availableVersion={softwareUpdateVersion}
+      updateVersion={updateVersion} />
 
     <hr />
+    <hr />
 
-    <h2 >Support Access and Factory Reset</h2>
+    <Button name='factory-reset' variant='danger' wide styleName='factory-reset-button' onClick={() => push('/factory-reset')}>Factory Reset</Button>
+  </PrimaryLayout>
+}
 
-    <SettingsFormInput
-      label='Access for HoloPort support (SSH)'
-      name='sshAccess'
-      type='checkbox'
-      checked={sshAccessVal}
-      onChange={handleToggleSshAccess} />
+export function SettingsTable ({ updateAvailable, header, children }) {
+  return <table styleName='settings-table' data-testid='settings-table'>
+    <thead>
+      <tr key='heading'>
+        <th id={header.toLowerCase().trim()} styleName='settings-row-header'>
+          <h5 styleName='row-header-title'>{header}
+            { updateAvailable
+              ? <span styleName='second-header'> Update Available</span>
+              : null
+            }
+          </h5>
+        </th>
+      </tr>
+    </thead>
+    <tbody>
+      {children}
+    </tbody>
+  </table>
+}
 
-    <Button name='factory-reset' variant='danger' wide onClick={() => push('/factory-reset')}>Factory Reset</Button>
-  </PrimaryLayout>
+export function SettingsRow ({ label, content, showSoftwareUpdateModal, updateAvailable, versionTable }) {
+  return <tr styleName='settings-row' data-testid='settings-row'>
+    <td styleName='settings-col align-left'>
+      <h3 styleName='row-label' data-testid='settings-label'>{label}</h3>
+    </td>
+    <td styleName='settings-col align-right'>
+      { updateAvailable && versionTable
+        ? <SoftwareUpdateButton availableVersion={content} showSoftwareUpdateModal={showSoftwareUpdateModal} />
+        : versionTable
+          ? <h3 styleName='row-content'>Your Software is up-to-date</h3>
+          : <h3 styleName='row-content'>{content}</h3>
+      }
+    </td>
+  </tr>
 }
 
-export function SettingsFormInput ({
-  name,
-  label,
-  type = 'number',
-  register,
-  errors = {},
-  ...inputProps
-}) {
-  return <>
-    {label && <label styleName='settingsLabel' htmlFor={name}>{label}</label>}
-    <Input name={name} id={name} type={type} placeholder={label} ref={register} {...inputProps} />
-    {errors[name] && <small styleName='field-error'>{errors[name].message}</small>}
-  </>
+function SoftwareUpdateButton ({ availableVersion, showSoftwareUpdateModal }) {
+  return <Button
+    onClick={() => showSoftwareUpdateModal(availableVersion)}
```

Comment:
yeah, I think it's a smell that we need to SoftwareUpdateButton to know `availableVersion`

line 146 should just read
```
onClick={showSoftwareUpdateModal}
```
or at worst, 
```
onClick={() => showSoftwareUpdateModal(true)}
```